src/pages/Medicine.jsx

```diff
@@ -0,0 +1,118 @@
+
+  const [itemData, setItemData] = useState({});
+
+  // const navigate = useNavigate();
+
+  const [medicineLog, setMedicineLog] = useLocalStorage("medicineLog", []);
+
+  const handleMedNameChange = (e) => {
+    setMedicineName(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedQtyOneChange = (e) => {
+    setMedicineQtyOne(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedDateChange = (e) => {
+    setMedicineDate(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedTimeChange = (e) => {
+    setMedicineTime(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedQtyTotalChange = (e) => {
+    setMedicineQtyTotal(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedDoseChange = (e) => {
+    setMedicineDose(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedIntervalChange = (e) => {
+    setMedicineInterval(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedStartDateChange = (e) => {
+    setMedicineStartDate(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleMedStartTimeChange = (e) => {
+    setMedicineStartTime(e.target.value);
+    setItemData({ ...itemData, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setMedicineLog([...medicineLog, itemData]);
+    setItemData({});
+    // navigate("/home");
+  };
+
+  // useEffect(() => {
+  //   localStorage.setItem("dataMedicine", JSON.stringify(dataMedicine));
+  // }, [dataMedicine]);
+
+  // console.log(dataMedicine)
+
+  return (
+    <div className="base-container-secondary">
+      <BackButton />
+      <p className="font-light w-screen mt-2 text-3xl text-dark-pink text-left pl-10">
+        Medicine
+      </p>
+      <FormMedicine
+        medicineName={medicineName}
+        medicineQtyOne={medicineQtyOne}
+        medicineDate={medicineDate}
+        medicineTime={medicineTime}
+        medicineQtyTotal={medicineQtyTotal}
+        medicineDose={medicineDose}
+        medicineInterval={medicineInterval}
+        medicineStartDate={medicineStartDate}
+        medicineStartTime={medicineStartTime}
+        onNameChange={handleMedNameChange}
+        onQtyOneChange={handleMedQtyOneChange}
+        onDateChange={handleMedDateChange}
+        onTimeChange={handleMedTimeChange}
+        onQtyTotalChange={handleMedQtyTotalChange}
+        onDoseChange={handleMedDoseChange}
+        onIntervalChange={handleMedIntervalChange}
+        onDateStartChange={handleMedStartDateChange}
+        onTimeStartChange={handleMedStartTimeChange}
```

Comment:
This confirms that we should just have a object for our medicine. This amount of props is unnecessary :D Ideally we got one object carrying all that data, and set certain properties as we need. 

```
const [medicine, setMedicine] = useState({ time: null, name: "", ...})
```

and then 

```
setMedicine({...medicine, name: event.target.value })
```

Reply:
```
const target = event.target;
const { value, name } = target;
setMedicine({...medicine, [name]: value })
```